inst/@sym/subsasgn.m

```diff
@@ -513,8 +513,78 @@
 %! a(1) = [];
 %! assert( isequal( a, b));
 
+%!test
+%! % indexing with a dimension more than 2
+%! a = sym ([1, 2; 3, 4; 5 6]);
+%! indices = [1 2; 3 4; 5 6];
+%! b = [10 11; 12 13; 14 15];
+%! a (indices) = b;
+%! assert (isequal (a, sym (b)));
+
 %!error <null assignment>
 %! a = sym([1, 2; 3, 4]);
 %! a(1, 2) = [];
 
+%!test
+%! A = sym ([0 0 0]);
+%! indices = [false true false];
+%! A (indices) = 1;
+%! assert (isequal (A, sym ([0 1 0])));
+%! A (indices) = [];
+%! assert (isequal (A, sym ([0 0])));
+%! indices = [false false];
+%! A (indices) = [];
+%! assert (isequal (A, sym ([0 0])));
+
+%!shared a, b
+%! a = [1 2 3 5;4 5 6 9;7 5 3 2];
+%! b = sym(a);
+
+%!function test_bool (a, b, c)
+%!  a(c) = 0;
+%!  b(c) = 0;
+%!  assert (nnz (a == 0), nnz (b == 0))
```

Comment:
Hi, note, i only choose nnz because there was 2 options, transform all the matrix to double, or count the number of zeros, i choose the last, if you think can help the other option no problem changing it.

Note 2, how the input matrix don't have any zero, count the number of zeros is the same as check the values changed in the matrices.

Thx. Cya.